core/common/src/serializers/DateTimePeriodSerializers.kt

```diff
@@ -0,0 +1,153 @@
+        PrimitiveSerialDescriptor("DateTimePeriod", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): DateTimePeriod =
+        DateTimePeriod.parse(decoder.decodeString())
+
+    override fun serialize(encoder: Encoder, value: DateTimePeriod) {
+        encoder.encodeString(value.toString())
+    }
+
+}
+
+object DatePeriodComponentSerializer: KSerializer<DatePeriod> {
+
+    private fun unexpectedNonzero(fieldName: String, value: Long) {
+        if (value != 0L) {
+            throw SerializationException("DatePeriod should have non-date components be zero, but got $value in '$fieldName'")
+        }
+    }
+
+    private fun unexpectedNonzero(fieldName: String, value: Int) = unexpectedNonzero(fieldName, value.toLong())
+
+    override val descriptor: SerialDescriptor =
+        buildClassSerialDescriptor("DatePeriod") {
+            element<Int>("years", isOptional = true)
+            element<Int>("months", isOptional = true)
+            element<Int>("days", isOptional = true)
+            element<Int>("hours", isOptional = true)
+            element<Int>("minutes", isOptional = true)
+            element<Int>("seconds", isOptional = true)
+            element<Long>("nanoseconds", isOptional = true)
+        }
+
+    override fun deserialize(decoder: Decoder): DatePeriod =
+        decoder.decodeStructure(descriptor) {
+            var years = 0
+            var months = 0
+            var days = 0
+            loop@while (true) {
+                when (val index = decodeElementIndex(descriptor)) {
+                    0 -> years = decodeIntElement(descriptor, 0)
+                    1 -> months = decodeIntElement(descriptor, 1)
+                    2 -> days = decodeIntElement(descriptor, 2)
+                    3 -> unexpectedNonzero("hours", decodeIntElement(descriptor, 3))
+                    4 -> unexpectedNonzero("minutes", decodeIntElement(descriptor, 4))
+                    5 -> unexpectedNonzero("seconds", decodeIntElement(descriptor, 5))
+                    6 -> unexpectedNonzero("nanoseconds", decodeLongElement(descriptor, 6))
+                    CompositeDecoder.DECODE_DONE -> break@loop // https://youtrack.jetbrains.com/issue/KT-42262
+                    else -> throw SerializationException("Unexpected index: $index")
+                }
+            }
+            DatePeriod(years, months, days)
+        }
+
+    override fun serialize(encoder: Encoder, value: DatePeriod) {
+        encoder.encodeStructure(descriptor) {
+            with(value) {
+                if (years != 0) encodeIntElement(DateTimePeriodComponentSerializer.descriptor, 0, years)
+                if (months != 0) encodeIntElement(DateTimePeriodComponentSerializer.descriptor, 1, months)
+                if (days != 0) encodeIntElement(DateTimePeriodComponentSerializer.descriptor, 2, days)
+            }
+        }
+    }
+
+}
+
+object DatePeriodISO8601Serializer: KSerializer<DatePeriod> {
```

Comment:
Not sure what you all use as a style guide, but I try really hard to evangelize the practice of treating acronyms and acronym-like things such as ISO, IPv6, iOS, etc. as words and casing them as such.

Google's Java and Kotlin style guide recommend this practice: 
* https://google.github.io/styleguide/javaguide.html#s5.3-camel-case
* https://developer.android.com/kotlin/style-guide#camel_case

That would make this type be `DatePeriodIso8601Serializer`. And I think there's one or two more in the change.

Reply:
+1 for DatePeriodIso8601Serializer

Reply:
Agree, it would be consistent with `Duration.toIsoString` in stdlib, for example.

What do you think about dropping `8601` suffix? On one hand, it's not that there are many ISO standards on date representation, on the other hand, the serializer name is not supposed to be used often, so the character economy there may be not worth it.